### PR TITLE
Mustang: Update rshim registers

### PR DESCRIPTION
**Pull request by asmaamellanox**
The Mustang rshim registers addresses are different from
BF1 and BF2 so we need to select that address based on the
platform. To achieve that, the rshim_regs struct has been
created and holds the value of all the registers' offsets.
when the rshim program is started, the rshim_regs struct
content is selected based on the value of bd->ver_id.
The masks and shifts are common to all platforms so there
is no need to add them to the struct, except for:
RSH_DEVICE_MSTR_PRIV_LVL__MEM_ACC_LVL_SHIFT
and
BF3_RSH_BOOT_FIFO_COUNT__BOOT_FIFO_COUNT_MASK

Note that I didn't make the corresponding changes to the PCIe
code. This needs to be addressed and tested in a different
patch.

### CODE REVIEW

**Review by lsun100 — APPROVED** [review]
Thanks for the changes. It looks good and extensible for new chips.

**Review by lsun100 — COMMENTED** [review]
Can we also update src/Makefile.am ?  (Please ignore  this comment)

File: src/rshim_regs.c [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#include "rshim_regs.h"
+#include "rshim.h"
+
+struct rshim_regs bf1_bf2_rshim_regs = {
```
[the remark this answers is not in the document]
Comment (hselasky) [reply src/rshim_regs.c:4]: What about using "const" keyword here?

File: src/rshim.h [hunk 2]
```diff
@@ -416,8 +416,39 @@ struct rshim_backend {
 
   /* API to enable the device. */
   int (*enable_device)(rshim_backend_t *bd, bool enable);
+
+  /* Platform specific register addresses */
+  struct rshim_regs *regs;
```
Comment (hselasky) [inline src/rshim.h:421]: const struct ....

**Pull request merged**